Support recycling ColumnChunkData during DROP TABLE

ericpolo opened the pull request:
1. Add logic to add FreeChunkEntry for possible recyclable ColumnChunkData from DROP TABLE
2. Support recycle ColumnChunkData from nested columns (String, Struct, List)
3. Add duplication check in FreeChunkMap
4. Fix comments left from previous PR #1

ericpolo committed:
Project: Fix Mayank's Comments

ericpolo committed:
Project: Support recycle Column Chunks during DROP TABLE (TESTED)

ericpolo committed:
Project: Extract chunk physical info based on inherit type of ColumnChunkData
  1. Support recycle ColumnChunkData nested in StringChunkData
  2. Support recycle ColumnChunkData nested in StructChunkData
  3. Support recycle ColumnChunkData nested in ListChunkData

ericpolo committed:
Project: Prevent add duplicate entries to FreeChunkMap

MayankJasoria commented on src/storage/store/node_group.cpp:
```diff
@@ -229,13 +229,19 @@ std::vector<std::pair<page_idx_t, page_idx_t>> NodeGroup::getAllChunkPhysicInfoF
+        } else {
+            ColumnChunkData &chunkData = chunkedGroup->getColumnChunk(columnID).getData();
+            std::vector<std::pair<page_idx_t, page_idx_t>> curInfo = chunkData.getAllChunkPhysicInfo();
+            chunkInfo.insert(chunkInfo.end(), curInfo.begin(), curInfo.end());
```
Lets move the common body of the if-else to a common function (besides 'i' and column id, the code is the same).
```
void addChunkDataForColumn(column_id_t columnID, std::vector<std::pair<page_idx_t, page_idx_t>> &chunkInfo) {
    ColumnChunkData &chunkData = chunkedGroup->getColumnChunk(columnID).getData();
     std::vector<std::pair<page_idx_t, page_idx_t>> curInfo = chunkData.getAllChunkPhysicInfo();
     chunkInfo.insert(chunkInfo.end(), curInfo.begin(), curInfo.end());
}
```

Additionally, instead of using INVALID_COLUMN_CHUNK as a parameter to run a different logic, can we have two separate functions? One that doesn't accept any inputs and returns information for all columns. Other one that only returns information for a specific column? Note that this also means the call to `getAllChunkPhysicInfo(INVALID_COLUMN_ID)` will also have to be updated.

MayankJasoria replied:
Addressed in updated patch, resolving.

ericpolo committed:
Project: Fix Mayank's Comments
  1. Introduce function getAllChunkPhysicInfo to resolve confusion
  2. Add function addChunkDataForColumn in NodeGroup
